### PR TITLE
Bug/issue 3370 - Unable to opt out of MetaMetrics (backup)

### LINKED ISSUE #3370: Unable to opt out of MetaMetrics
**Describe the bug**
I am unable to opt out of Metametrics from the security & privacy view.  

**Screenshots**
See slack conversation: https://consensys.slack.com/archives/G8RSKCNCD/p1635266451061600

Also see recording [here](http://recordit.co/NZk5hPr0ap)

**To Reproduce**
1.  Launch the app. 
2. Opt in for metametrics
3. Create/import your wallet
4. Tap the burger menu
5. Go to the settings view -> security & privacy -> metametrics.  toggle metametrics to "Off"
6. Kill the app then relaunch it
7. Tap the burger menu
8. Go to the settings page -> security & privacy -> metametrics.
9. Notice metametrics is still toggled on. 

**Expected behavior**
A user should be able to opt out of Metametrics.  

**Smartphone (please complete the following information):**
 - App Version [e.g. 1.0.0] - find version number in app from Settings > About MetaMask

-------------------------------------------------------------
_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user?
- Add stats if available on % of customers impacted
- Is this visible to all users?
- Is this tech debt?

### PR DESCRIPTION
**Pull request by Fatxx**
**Description**

Fixing issue where Metametrics option in settings was not persisted between application sessions. 

**Checklist**

* [X] There is a related GitHub issue
* [ ] Tests are included if applicable
* [X] Any added code is fully documented

**Issue**

Resolves #3370

### CODE REVIEW

**Comment by Fatxx** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Fatxx** [comment]
@sethkfman basically, I moved the flag to the redux store and removed the `analytics` store from the blacklist to persist the value.

**Comment by sethkfman** [comment]
@Fatxx We had added analytics to the redux persist blacklist because we are trying to move towards using more native stores for persisting simple data. I think you solution would work great but we want to move away from redux for these simple persisted data types. 

We are using DefaultPreferences to persist this value and must have missed this update. This is were the metrics are first set when the user onboards: https://github.com/MetaMask/metamask-mobile/blob/1647241bd25d7b5e38400bfa10a66e01f41c6dd8/app/components/UI/OptinMetrics/index.js#L207

I think we just need to initialize this value in the view and make sure the Analytics class also uses it.

**Comment by Fatxx** [comment]
@sethkfman  A question that arises on my side is that you're alerting the user that a restart is required. However, as far as I'm concerned, it seems that the tracking is disabled right after `Analytics.disable()` is executed since the flag inside the `Analytics` instance changes to `false,` meaning that the `_trackEvent` never fires an `RCTAnalytics.trackEvent` anymore.

Is there any hidden reason for this that I'm not aware of?

**Comment by Fatxx** [comment]
@sethkfman just pushed new changes using `DefaultPreference` instead of Redux. 

To sum up, I basically removed `Analytics` from Redux since we now rely only on DefaultPreference as the source of truth, this preference is loaded on Analytics initialization which caches the flag value in memory in the class instance. I also removed the AnalyticsService since in my POV is redundant because we can simply init the Analytics directly.

I removed the user alert for now since it's a false positive of what's actually happening.

File: app/components/Views/Settings/SecuritySettings/index.js [hunk 1]
```diff
@@ -460,16 +459,12 @@ class Settings extends PureComponent {
...
 		if (value) {
 			Analytics.enable();
-			this.setState({ metricsOptIn: true });
+			this.setState({ analyticsEnabled: true });
 			await this.trackOptInEvent('Metrics Opt In');
 		} else {
 			await this.trackOptInEvent('Metrics Opt Out');
 			Analytics.disable();
-			this.setState({ metricsOptIn: false });
-			Alert.alert(
```
Comment (sethkfman) [inline app/components/Views/Settings/SecuritySettings/index.js:469]: @Fatxx I spoke with Andre and we are pretty sure we need this Alert due some native initialization. I would keep the Alert for now.

**Review by sethkfman — COMMENTED** [review]
Minor comment on the Alert

File: app/components/Views/Settings/SecuritySettings/index.js (hunk 1 above)
Comment (Fatxx) [reply app/components/Views/Settings/SecuritySettings/index.js:469]: Ok sure, I'll add that.

**Pull request closed without merging**